libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -617,3 +618,44 @@ void AP_Baro::set_pressure_correction(uint8_t instance, float p_correction)
         sensors[instance].p_correction = p_correction;
     }
 }
+
+/*
+ * register new baro if there is space for it
+ */
+bool AP_Baro::register_uavcan_baro(uint8_t mgr, uint8_t node)
+{
+#if HAL_WITH_UAVCAN
+    // Check if this node in network is already registered
+    for (uint8_t i = 0; i < _num_drivers; i++) {
+        if (sensors[i].can_manager == mgr && sensors[i].uavcan_node == node) {
+            return true;
+        }
+    }
+
+    // If there is place left - allocate one UAVCAN based baro
+    if (_num_drivers < BARO_MAX_DRIVERS && _num_sensors < BARO_MAX_INSTANCES)
+    {
+        if ((AP_BoardConfig_CAN::get_can_enable() != 0) && (hal.can_mgr[mgr] != nullptr))
+        {
+            drivers[_num_drivers] = new AP_Baro_UAVCAN(*this);
+
+            if (((AP_Baro_UAVCAN *) drivers[_num_drivers])->register_uavcan_baro(mgr, node)) {
```

Comment:
I'm not sure how else to explain it. This has nothing to do with type or delay. I'm just asking this code to be moved inside the AP_Baro_UAVCAN class. Look at one of the drivers: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Baro/AP_Baro_MS5611.cpp#L63-L76

The init method would have what you have in your `register_uavcan_baro`.